windows/src/test/unit-tests/androidstringtokeymanlocalestring/Keyman.Test.AndroidStringToKeymanLocaleStringTest.pas

```diff
@@ -0,0 +1,37 @@
+unit Keyman.Test.AndroidStringToKeymanLocaleStringTest;
+
+interface
+
+uses
+  DUnitX.TestFramework,
+  Keyman.System.AndroidStringToKeymanLocaleString;
+
+type
+  [TestFixture]
+  TAndroidStringToKeymanLocaleStringTest = class(TObject)
+  public
+    [Test]
+    procedure TestTransform;
+  end;
+
+implementation
+
+{ TAndroidStringToKeymanLocaleStringTest }
+
+procedure TAndroidStringToKeymanLocaleStringTest.TestTransform;
+begin
+  Assert.AreEqual('', TAndroidStringToKeymanLocaleString.Transform(''));
+  Assert.AreEqual('text', TAndroidStringToKeymanLocaleString.Transform('text'));
+  Assert.AreEqual('text', TAndroidStringToKeymanLocaleString.Transform('"text"'));
+  Assert.AreEqual('te"xt', TAndroidStringToKeymanLocaleString.Transform('te\"xt'));
+  Assert.AreEqual('te'#13#10'xt', TAndroidStringToKeymanLocaleString.Transform('te\nxt'));
+  Assert.AreEqual('1\2?3@4', TAndroidStringToKeymanLocaleString.Transform('1\\2\?3\@4'));
+  Assert.AreEqual('%2:s text %0:d %1:s', TAndroidStringToKeymanLocaleString.Transform('%3$s text %1$d %2$s'));
```

Comment:
What about testing other formats?
~'d'~, 'u', 'e', 'f', 'g', 'n', 'm', 'p', ~'s'~, 'x'

Reply:
I don't think we have to check every flavour -- it doesn't improve the quality of the test case, really.